core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/ColumnPageEncoder.java

```diff
@@ -147,10 +160,48 @@ protected void fillLegacyFields(DataChunk2 dataChunk)
 
   public static EncodedColumnPage encodedColumn(ColumnPage page)
       throws IOException, MemoryException {
-    ColumnPageEncoder encoder = new DirectCompressCodec(DataTypes.BYTE_ARRAY).createEncoder(null);
-    return encoder.encode(page);
+    ColumnPageEncoder pageEncoder = createCodecForDimension(page);
```

Comment:
FallbackEncodedColumnPage is also using complex type so child column types loop logic not present.

Reply:
Looping logic is not required as only one particular column will only come during execution. No need to further drill down to its children.

Reply:
Added Unsupported Exception for Non Complex Primitive.